### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by skrawcz**
Tweaks wording in README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,10 +32,10 @@ including tooling to build a UI in streamlit and watch your state machine execut
 
 ## Why the name Burr?
 
-Burr is named after Aaron Burr, founding father, third VP of the United States, and murderer/arch-nemesis of Alexander Hamilton.
-We imagine a world in which Burr and Hamilton lived in harmony and saw through their differences. We originally
-built Burr as a _harness_ to handle state between executions of Hamilton DAGs,
-but realized that it has a wide array of applications and decided to release it.
+Burr is named after [Aaron Burr](https://en.wikipedia.org/wiki/Aaron_Burr), founding father, third VP of the United States, and murderer/arch-nemesis of [Alexander Hamilton](https://en.wikipedia.org/wiki/Alexander_Hamilton).
+In our version of things here, we imagine a world in which Burr and Hamilton lived in harmony and saw through their differences and thus were happy to work together. We originally
```
Comment (elijahbenizzy) [inline README.md:37]: "In our version of things here" -- unecessary verbosity -- implied by "we imagine"

File: README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Burr can be used for a variety of applications. Burr can build a state machine t
 And a lot more!
 
 Using hooks and other integrations you can (a) integrate with any of your favorite vendors (LLM observability, storage, etc...), and
-(b) build custom actions that delegate to your favorite libraries.
+(b) build custom actions that delegate to your favorite libraries like [Hamilton](github.com/DAGWorks-Inc/hamilton).
```
Comment (elijahbenizzy) [inline README.md:25]: (like hamilton) -- in parens

File: README.md [hunk 3]
```diff
@@ -2,14 +2,14 @@
...
+Burr is a way to express a state machine (i.e. a graph/flowchart) for data/AI projects. You can (and should!) use it for anything where managing state can be hard. Hint: managing state
+is always hard! Like in a RAG chat application, or if you're doing hyperparameter tuning, or a trading simulation.
 
 Link to [documentation](https://burr.dagworks.io/). Quick video intro [here](https://www.loom.com/share/8a92474bb7574d6eb4cd25c21913adf2).
 
 Burr is:
-1. A (dependency-free) python library that enables you to build and manage state machines with simple python functions
-2. A server that can run your state machines and provide a UI to track execution/debug
+1. A (dependency-free) low abstraction python library that enables you to build and manage state machines with simple python functions
+2. It optionally, comes with a server to view execution telemetry for introspection and debugging
```
Comment (elijahbenizzy) [inline README.md:12]: Remove comma after "optionally"

File: README.md [hunk 4]
```diff
@@ -32,10 +32,10 @@ including tooling to build a UI in streamlit and watch your state machine execut
 
 ## Why the name Burr?
 
-Burr is named after Aaron Burr, founding father, third VP of the United States, and murderer/arch-nemesis of Alexander Hamilton.
-We imagine a world in which Burr and Hamilton lived in harmony and saw through their differences. We originally
-built Burr as a _harness_ to handle state between executions of Hamilton DAGs,
-but realized that it has a wide array of applications and decided to release it.
+Burr is named after [Aaron Burr](https://en.wikipedia.org/wiki/Aaron_Burr), founding father, third VP of the United States, and murderer/arch-nemesis of [Alexander Hamilton](https://en.wikipedia.org/wiki/Alexander_Hamilton).
+In our version of things here, we imagine a world in which Burr and Hamilton lived in harmony and saw through their differences and thus were happy to work together. We originally
+built Burr as a _harness_ to handle state between executions of Hamilton DAGs (because DAGs don't have cycles),
```
Comment (elijahbenizzy) [inline README.md:38]: Don't need to mention DAGs + cycles its confusing. DAGs don't carry state.

File: README.md (hunk 3 above)
Comment (skrawcz) [reply README.md:12]:
```suggestion
2. It optionally comes with a server to view execution telemetry for introspection and debugging
```

File: README.md [hunk 5]
```diff
@@ -2,14 +2,14 @@
...
+Burr is a way to express a state machine (i.e. a graph/flowchart) for data/AI projects. You can (and should!) use it for anything where managing state can be hard. Hint: managing state
+is always hard! Like in a RAG chat application, or if you're doing hyperparameter tuning, or a trading simulation.
 
 Link to [documentation](https://burr.dagworks.io/). Quick video intro [here](https://www.loom.com/share/8a92474bb7574d6eb4cd25c21913adf2).
 
 Burr is:
-1. A (dependency-free) python library that enables you to build and manage state machines with simple python functions
-2. A server that can run your state machines and provide a UI to track execution/debug
+1. A (dependency-free) low abstraction python library that enables you to build and manage state machines with simple python functions
+2. It optionally comes with a server to view execution telemetry for introspection and debugging
```
Comment (skrawcz) [inline README.md:12]:
```suggestion
2. It optionally comes with a UI to view execution telemetry for introspection and debugging
```

File: README.md [hunk 6]
```diff
@@ -2,14 +2,14 @@
...
+Burr is a way to express a state machine (i.e. a graph/flowchart) for data/AI projects. You can (and should!) use it for anything where managing state can be hard. Hint: managing state
+is always hard! Like in a RAG chat application, or if you're doing hyperparameter tuning, or a trading simulation.
 
 Link to [documentation](https://burr.dagworks.io/). Quick video intro [here](https://www.loom.com/share/8a92474bb7574d6eb4cd25c21913adf2).
 
 Burr is:
-1. A (dependency-free) python library that enables you to build and manage state machines with simple python functions
-2. A server that can run your state machines and provide a UI to track execution/debug
+1. A (dependency-free) low abstraction python library that enables you to build and manage state machines with simple python functions
+2. It optionally comes with a UI to view execution telemetry for introspection and debugging
```
Comment (skrawcz) [inline README.md:12]:
```suggestion
2. It comes with a UI you can use view execution telemetry for introspection and debugging
```

File: README.md (hunk 2 above)
Comment (skrawcz) [inline README.md:25]:
```suggestion
(b) build custom actions that delegate to your favorite libraries (like [Hamilton](github.com/DAGWorks-Inc/hamilton)).
```

File: README.md [hunk 7]
```diff
@@ -32,10 +32,11 @@ including tooling to build a UI in streamlit and watch your state machine execut
 
 ## Why the name Burr?
 
-Burr is named after Aaron Burr, founding father, third VP of the United States, and murderer/arch-nemesis of Alexander Hamilton.
-We imagine a world in which Burr and Hamilton lived in harmony and saw through their differences. We originally
-built Burr as a _harness_ to handle state between executions of Hamilton DAGs,
-but realized that it has a wide array of applications and decided to release it.
+Burr is named after [Aaron Burr](https://en.wikipedia.org/wiki/Aaron_Burr), founding father, third VP of the United States, and murderer/arch-nemesis of [Alexander Hamilton](https://en.wikipedia.org/wiki/Alexander_Hamilton).
+What's the connection with Hamilton? This is [DAGWorks](www.dagworks.io)' second open-source library release after the [Hamilton library](https://github.com/dagworks-inc/hamilton)
+In our version of things here, we imagine a world in which Burr and Hamilton lived in harmony and saw through their differences and thus were happy to work together. We originally
```
Comment (skrawcz) [inline README.md:37]:
```suggestion
Here we imagine a world in which Burr and Hamilton lived in harmony and saw through their differences and thus were happy to work together. We originally
```

**Pull request merged**